src/providers/account.tsx

```diff
@@ -1,59 +1,87 @@
-import React, { createContext, useEffect, useMemo, useState } from 'react';
-import { useApi, useAssets } from '../hooks';
-import { Asset, IAccountMeta } from '../model';
-import { convertToSS58, isSameAddress, readStorage, updateStorage } from '../utils';
+import { createContext, useEffect, useState, useCallback } from 'react';
+import keyring from '@polkadot/ui-keyring';
+import { useAssets, useWallet } from '../hooks';
+import { Asset, Account } from '../model';
+import { readStorage, updateStorage } from '../utils';
+import { SEARCH_PARAMS_SOURCE } from '../config';
+import { SelectAccountModal } from '../components/widget/account/SelectAccountModal';
 
 export interface AccountCtx {
-  account: string;
-  setAccount: (account: string) => void;
-  accountWithMeta: IAccountMeta;
   assets: Asset[];
-  getBalances: (acc?: string) => void;
-}
+  assetsLoading: boolean | undefined;
+  account: Account | null | undefined;
 
-const DEFAULT_ADDRESS_PREFIX = 42; // Substrate, 42
+  refreshAssets: () => void;
+  selectAccount: (address: string) => void;
+}
 
-export const AccountContext = createContext<AccountCtx | null>(null);
+export const AccountContext = createContext<AccountCtx>({} as AccountCtx);
 
 export const AccountProvider = ({ children }: React.PropsWithChildren<unknown>) => {
-  const [account, setAccount] = useState<string>('');
-  const { network, connection } = useApi();
-  const accountWithMeta = useMemo(
-    () => connection.accounts.find((item) => isSameAddress(item.address, account)) ?? connection.accounts[0],
-    [account, connection]
+  const { accounts } = useWallet();
+  const [account, setAccount] = useState<Account | null>();
+  const { assets, loading: assetsLoading, getAssets: refreshAssets } = useAssets(account?.displayAddress || '');
+  const [visible, setVisible] = useState(false);
+
+  const selectAccount = useCallback(
+    (address: string) => {
+      setAccount(accounts.find((acc) => acc.address === address));
+    },
+    [accounts]
```

Comment:
Did the editor report no type error of setAccount?

Reply:
no, why would it report error

Reply:
accounts.find((acc) => acc.address === address) :  Account | undefined. declared type：Account | null

Reply:
declared type：Account | null | undefine

<img width="692" alt="image" src="https://user-images.githubusercontent.com/21007682/173479388-59f5bfb0-2d71-46a8-ab59-db235c4e02cf.png">

did you report an error there